### PR TITLE
feat: add option to include system fields in download

### PR DESCRIPTION
**Pull request by connoratrug**
Add adding a param `includeSystemColumns=true' to the download api request, the system columns are added to the download.

`/pet store/api/csv/Pet` (without system columns)
`/pet store/api/csv/Pet?includeSystemColumns=true` (with system columns)

found issue with exel not working 
still need to add general dowload  tests for all api types

### CODE REVIEW

**Review by mswertz — CHANGES_REQUESTED** [review]
Looks good.

Two comments:
* can you also update the user interface so there is an option to use this (see screenshot)? Or do you plan to do this immediately thereafter
* test coverage is <80%, maybe that was already the case for these parts but can you please check?

image

We could simply add additional links, or a bit more advanced by having a checkbox that influences al the hyperlinks to include the parameter.

Also off topic: currently this will include all 'draft' records, correct? And in terms of defaults, don't we prefer to always include metadata? What do data managers say?

**Comment by connoratrug** [comment]
> * can you also update the user interface so there is an option to use this (see screenshot)? Or do you plan to do this immediately thereafter

Would suggest to keep this as API only option.
Could add it to UI ( suggest separate PR ) but this would needless complicate the UI imo.


> We could simply add additional links, or a bit more advanced by having a checkbox that influences al the hyperlinks to include the parameter.

prefer the additional links option, but not sure if adding it to the ui is a tradeoff thats worth it ( extra option versus ui complexity) 

I general i could argue against adding a 'mode' checkbox, generally modes in ui are confusing ( example https://www.amazon.com/Set-Phasers-Stun-Design-Technology/dp/0963617885). Emx1 had a similar mode function that was confusing. Would suggest keeping this a 'expert' function ( or switch it to default it this is the preferred mode)


> * test coverage is <80%, maybe that was already the case for these parts but can you please check?

Yes additional tests for excel and zip would be good, but when i try to read the downloaded exel file in the test ( via TableStoreForXlsxFile0 i get ```org.apache.poi.openxml4j.exceptions.NotOfficeXmlFileException: No valid entries or contents found, this is not a valid OOXML (Office Open XML) file``` i asked about this in the emx2-support channel. 

> 
> Also off topic: currently this will include all 'draft' records, correct? And in terms of defaults, don't we prefer to always include metadata? What do data managers say?

This does not change the row ( only the columns ), we should ask data managers about preferred default, however data managers are not the only usergroup.

I think current default ( no system columns , and include draft rows ) is a good default, if we do add extra options i would suggest adding a options to include/exclude draf row instead of including/excluding system columns ( seems more like a 'expert' option to me).

**Comment by mswertz** [comment]
Ok, then leave the 'UI' untouched for now, I noticed you did update docs so interested users will know about this.

**Review by BrendaHijmans — COMMENTED** [review]
functional review: up- and download of system fields works

File: .gitignore [hunk 1]
```diff
@@ -45,5 +45,9 @@ tls
 yarn_error.log
 .nyc_output
 coverage
+
+/.run/Run Molgenis .run.xml
```
Comment (mswertz) [inline .gitignore:49]: what is this?

File: backend/molgenis-emx2-webapi/src/test/java/org.molgenis.emx2.web/TestUtils.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import java.util.Iterator;
+import java.util.List;
+import net.lingala.zip4j.io.inputstream.ZipInputStream;
+import net.lingala.zip4j.model.LocalFileHeader;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.xssf.usermodel.XSSFSheet;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+
+public class TestUtils {
```
Comment (mswertz) [inline backend/molgenis-emx2-webapi/src/test/java/org.molgenis.emx2.web/TestUtils.java:20]: I am a bit surprised you didn't use the TableStore classes here.

**Review by mswertz — APPROVED** [review]

File: backend/molgenis-emx2-webapi/src/test/java/org.molgenis.emx2.web/TestUtils.java (hunk 2 above)
Comment (connoratrug) [reply backend/molgenis-emx2-webapi/src/test/java/org.molgenis.emx2.web/TestUtils.java:20]: TableStore does not allow for easy parsing of sheet to String list ( only via Models objects). Having a separate test util makes it easy to write some tests helpers without the need to complicate the application code. Could add similar methods to application code, but if these methods are only there as test helpers i would argue against this, as is makes the api more complex for the caller

**Pull request merged**